client/src/components/Applications/AppForm/AppForm.js

```diff
@@ -67,15 +72,15 @@ const AppDetail = () => {
     );
   }
 
-  if (loading || !app.name || allocationsLoading) {
+  if (loading || allocationsLoading) {
     return <LoadingSpinner />;
   }
 
   return (
     <div id="appDetail-wrapper">
-      {!app.name && <AppPlaceholder />}
-      {app.value && app.value.type === 'html' ? (
-        parse(app.value.definition.html)
+      {!app && <AppPlaceholder apps={hasApps} />}
+      {app.html ? (
```

Comment:
Think this should be checking `app.appType` to avoid cases where the `html` field is accidentally filled in for an `agave` app
```suggestion
      {app.appType === 'html' ? (
```